staging/src/k8s.io/apiserver/pkg/util/webhook/serviceresolver.go

```diff
@@ -38,9 +38,9 @@ func NewDefaultServiceResolver() ServiceResolver {
 // note that the name and namespace are required and by default all created addresses use HTTPS scheme.
 // for example:
 //  name=ross namespace=andromeda resolves to https://ross.andromeda.svc:443
-func (sr defaultServiceResolver) ResolveEndpoint(namespace, name string) (*url.URL, error) {
+func (sr defaultServiceResolver) ResolveEndpoint(namespace, name string, port int32) (*url.URL, error) {
 	if len(name) == 0 || len(namespace) == 0 {
```

Comment:
add `|| port == 0` and change error message to `cannot resolve an empty service name or namespace or port`